marconi-chain-index/src/Marconi/ChainIndex/Api/JsonRpc/Endpoint/Utxo.hs

```diff
@@ -0,0 +1,157 @@
+ )
+import Marconi.ChainIndex.Api.Types (HttpServerConfig, configQueryables, configTrackedAddresses)
+import Marconi.ChainIndex.Indexers (queryableUtxo)
+import Marconi.ChainIndex.Indexers.BlockInfo qualified as BI
+import Marconi.ChainIndex.Indexers.Utxo (datumHash, txIn, txIndex, value)
+import Marconi.ChainIndex.Indexers.UtxoQuery (
+  UtxoQueryInput (UtxoQueryInput),
+  UtxoResult (UtxoResult),
+ )
+import Marconi.Core qualified as Core
+import Marconi.Core.JsonRpc (ReaderHandler, hoistHttpHandler, queryErrToRpcErr)
+import Marconi.Core.Type (Result)
+import Network.JsonRpc.Types (JsonRpc, JsonRpcErr, mkJsonRpcInternalErr, mkJsonRpcInvalidParamsErr)
+import Servant.Server (Handler)
+
+------------------
+-- Method types --
+------------------
+
+type RpcGetUtxosFromAddressMethod =
+  JsonRpc
+    "getUtxosFromAddress"
+    GetUtxosFromAddressParams
+    String
+    GetUtxosFromAddressResult
+
+--------------
+-- Handlers --
+--------------
+
+-- | Handler for retrieving UTXOs by Address
+getUtxoQueryInputHandler
+  :: UtxoQueryInput
+  -- ^ Bech32 addressCredential and possibly a pair of slotNumbers
+  -> ReaderHandler HttpServerConfig (Either (JsonRpcErr String) (Result UtxoQueryInput))
+getUtxoQueryInputHandler query = do
+  indexer <- view (configQueryables . queryableUtxo)
+  let sqliteAggregateLastSyncPoint :: ExceptT (Core.QueryError UtxoQueryInput) Handler C.ChainPoint
+      sqliteAggregateLastSyncPoint = do
+        res <- liftIO $ runExceptT $ lift $ Core.lastSyncPoint indexer
+        case res of
+          Left err -> throwError err
+          Right res' -> pure res'
+  lastPointM <- hoistHttpHandler $ runExceptT sqliteAggregateLastSyncPoint
+  case lastPointM of
+    Left err ->
+      pure $
+        Left $
+          mkJsonRpcInternalErr $
+            Just $
+              "Can't resolve last point in getUtxosFromAddress: " <> show err
+    Right lastPoint ->
+      hoistHttpHandler $
+        liftIO $
+          first queryErrToRpcErr <$> Core.queryEither lastPoint query indexer
+
+-- | Return 'GetBurnTokenEventsResult' based on 'GetBurnTokenEventsParams'
+getUtxosFromAddressQueryHandler
+  :: GetUtxosFromAddressParams
+  -> ReaderHandler HttpServerConfig (Either (JsonRpcErr String) GetUtxosFromAddressResult)
+getUtxosFromAddressQueryHandler query = do
+  addresses <- view configTrackedAddresses
+  res <- join <$> traverse getUtxoQueryInputHandler (mapGetBurnTokenEventsQuery addresses query)
+  pure $ mapGetBurnTokenEventsResult =<< res
+  where
+    mapGetBurnTokenEventsQuery
+      :: [C.AddressAny] -> GetUtxosFromAddressParams -> Either (JsonRpcErr String) UtxoQueryInput
+    mapGetBurnTokenEventsQuery
+      _ -- targetAddresses
+      (GetUtxosFromAddressParams addr createdAtAfter unspentBefore) = do
+        mappedAddr <- addressMapping (pack addr)
+        -- TODO I can't actually see where we used to do this, and I get this error when running `run-pre-prod-queries`. Am I missing something?
```

Comment:
Note that the reason for this TODO is the spec references an error `The 'address' param value must belong to the provided target addresses.`, but I can't see where we were surfacing this previously. I implemented the logic that I presumed would be the source of the error, but it caused the error to always be thrown with my current setup.

Reply:
The filtering list can be passed when we start `marconi-chain-index`, it's the one exposed by the `TargetAddresses` endpoint. If no list is given, we index all the addresses and can query them all.